src/langchain/llm-chat.ts

```diff
@@ -3,117 +3,175 @@ import {
+          : { model_id: this.modelId, prompt_id: this.promptId }),
+        messages: this._convertMessages(messages),
+        parameters: merge(this.parameters, options.parameters),
+      },
+      { signal: options.signal },
+    );
+    if (output.results.length !== 1) throw new InternalError('Invalid result');
+    const result = output.results[0];
+    if (result.input_token_count == null)
+      throw new InternalError('Missing token count');
+    this.conversation = output.conversation_id;
     return {
-      generations: output.generations.map(([generation]) => ({
-        message: new SystemMessage(generation.text),
-        generationInfo: generation.generationInfo,
-        text: generation.text,
-      })),
-      llmOutput: output.llmOutput,
+      generations: [
+        {
+          message: new AIMessage({ content: result.generated_text }),
+          text: result.generated_text,
+          generationInfo: {
+            inputTokens: result.input_tokens,
+            generatedTokens: result.generated_tokens,
+            seed: result.seed,
+            stopReason: result.stop_reason,
+            stopSequence: result.stop_sequence,
+            moderation: result.moderation,
+          },
+        },
+      ],
+      llmOutput: {
+        tokenUsage: {
+          completionTokens: result.generated_token_count,
+          promptTokens: result.input_token_count,
+          totalTokens: result.generated_token_count + result.input_token_count,
+        },
+      },
     };
   }
 
-  _combineLLMOutput(...llmOutputs: TextGenerationCreateOutput[]) {
-    return llmOutputs
-      .flatMap((output) => output.results?.at(0) ?? [])
-      .reduce(
-        (acc, gen) => {
-          acc.tokenUsage.completionTokens += gen.generated_token_count || 0;
-          acc.tokenUsage.promptTokens += gen.input_token_count || 0;
-          acc.tokenUsage.totalTokens =
-            acc.tokenUsage.promptTokens + acc.tokenUsage.completionTokens;
-
-          return acc;
-        },
-        {
-          tokenUsage: {
-            completionTokens: 0,
-            promptTokens: 0,
-            totalTokens: 0,
-          },
+  async *_streamResponseChunks(
+    messages: BaseMessage[],
+    options: this['ParsedCallOptions'],
+    runManager?: CallbackManagerForLLMRun,
+  ): AsyncGenerator<ChatGenerationChunk> {
+    const outputStream = await this.client.text.chat.create_stream(
+      {
+        ...(this.conversation
+          ? { conversation_id: this.conversation }
+          : { model_id: this.modelId, prompt_id: this.promptId }),
+        messages: this._convertMessages(messages),
+        parameters: merge(this.parameters, options.parameters),
+      },
+      { signal: options.signal }
+    );
+    for await (const output of outputStream) {
+      this.conversation = output.conversation_id;
+      if (!output.results) continue;
+      if (output.results.length !== 1)
+        throw new InternalError('Invalid output');
```

Comment:
This is not correct handling and is extremely restrictive.

In case of multiple results, just yield all of them.
In case of moderation result, return an empty message with `generationInfo`, which sends the `moderation`.


⚠️ Note: inspect response chunkes, `result.moderation` is `undefined` if there is a text result.